tests/test-update-refspec-checkpoint.c

```diff
@@ -0,0 +1,238 @@
+    *out_files_hashtable = g_hash_table_new_full (g_direct_hash,
+                                                  g_direct_equal,
+                                                  NULL,
+                                                  (GDestroyNotify) g_ptr_array_free);
+
+  directories_to_create_strv[0] = g_build_filename ("usr",
+                                                    "share",
+                                                    "eos-updater",
+                                                    NULL);
+  directories_to_create_strv[1] = NULL;
+  g_ptr_array_add (files_to_create,
+                   simple_file_new_steal (g_build_filename ("usr",
+                                                            "share",
+                                                            "eos-updater",
+                                                            "upgrade-refspec",
+                                                            NULL),
+                                          g_strdup (new_refspec)));
+
+  g_hash_table_insert (*out_directories_hashtable,
+                       GUINT_TO_POINTER (commit),
+                       g_steal_pointer (&directories_to_create_strv));
+  g_hash_table_insert (*out_files_hashtable,
+                       GUINT_TO_POINTER (commit),
+                       g_steal_pointer (&files_to_create));
+}
+
+static void
+update_client (EosUpdaterFixture *fixture,
+               EosTestClient     *client)
+{
+  DownloadSource main_source = DOWNLOAD_MAIN;
+  g_autoptr(GVariant) main_source_variant = NULL;
+  g_autoptr(GPtrArray) cmds = NULL;
+  g_auto(CmdAsyncResult) updater_cmd = CMD_ASYNC_RESULT_CLEARED;
+  g_autoptr(GFile) autoupdater_root = NULL;
+  g_autoptr(EosTestAutoupdater) autoupdater = NULL;
+  g_auto(CmdResult) reaped = CMD_RESULT_CLEARED;
+  g_autoptr(GError) error = NULL;
+
+  eos_test_client_run_updater (client,
+                               &main_source,
+                               &main_source_variant,
+                               1,
+                               &updater_cmd,
+                               &error);
+  g_assert_no_error (error);
+
+  autoupdater_root = g_file_get_child (fixture->tmpdir, "autoupdater");
+  autoupdater = eos_test_autoupdater_new (autoupdater_root,
+                                          UPDATE_STEP_APPLY,
+                                          1,
+                                          TRUE,
+                                          &error);
+  g_assert_no_error (error);
+
+  eos_test_client_reap_updater (client,
+                                &updater_cmd,
+                                &reaped,
+                                &error);
+  g_assert_no_error (error);
+
+  cmds = g_ptr_array_new ();
+  g_ptr_array_add (cmds, &reaped);
+  g_ptr_array_add (cmds, autoupdater->cmd);
+  g_assert_true (cmd_result_ensure_all_ok_verbose (cmds));
+}
+
+static void
+test_update_refspec_checkpoint (EosUpdaterFixture *fixture,
```

Comment:
As with all test cases, please add a comment above the test case function which explains roughly what it tries to test and how it goes about that.